osutil/digest.go

```diff
@@ -44,3 +44,22 @@ func FileDigest(filename string, hash crypto.Hash) ([]byte, uint64, error) {
 	}
 	return h.Sum(nil), uint64(size), nil
 }
+
+// PartialFileDigest computes a hash digest of the file starting from an offset using the given hash.
+// It also returns the size of the data that were hashed.
+func PartialFileDigest(filename string, hash crypto.Hash, offset uint64) ([]byte, uint64, error) {
+	f, err := os.Open(filename)
+	if err != nil {
+		return nil, 0, err
+	}
+	defer f.Close()
+
```

Comment:
I wonder if we should add a check that offset is smaller than the size of the file

Reply:
I think Seek does lseek which should return EINVAL if something goes wrong there. In any case I've added one. 

By the way I've added `PartialFileDigest` as a separate function to make the intention clear for now but maybe it would be even better to combine both functions in one. With the extra check there, that would mean that if I had `FileDigest` call `PartialFileDigest` with a 0 offset, `FileDigest` would call an extra `Stat` + a comparison. Would that be bad for performance?